packages/x-charts/src/internals/plugins/featurePlugins/useChartHighlight/useChartHighlight.ts

```diff
@@ -18,27 +19,48 @@ export const useChartHighlight: ChartPlugin<UseChartHighlightSignature> = ({ sto
     if (store.state.highlight.item !== params.highlightedItem) {
       store.set('highlight', { ...store.state.highlight, item: params.highlightedItem });
     }
+    if (process.env.NODE_ENV !== 'production') {
+      // Warn if controlled but onHighlightChange is not provided
+      if (params.highlightedItem !== undefined && !store.state.highlight.isControlled) {
+        warnOnce(
+          [
+            'MUI X Charts: The `highlightedItem` switched between controlled and uncontrolled state.',
+            'To remove highlight in control mode, you must provide `null` as `highlightedItem` value and not `undefined`.',
+          ].join('\n'),
+        );
+      }
+    }
   }, [store, params.highlightedItem]);
 
   const clearHighlight = useEventCallback(() => {
     params.onHighlightChange?.(null);
-    const prevItem = store.getSnapshot().highlight.item;
-    if (prevItem === null) {
+    const prevHighlight = store.getSnapshot().highlight;
+    if (prevHighlight.item === null || prevHighlight.isControlled) {
       return;
     }
 
-    store.set('highlight', { item: null, lastUpdate: 'pointer' });
+    store.set('highlight', {
+      ...prevHighlight,
+      item: null,
+      lastUpdate: 'pointer',
+      isControlled: false,
```

Comment:
Why do we change this to false? 🤔

Reply:
In the version you've seen few minutes ago, there were a `...prevHighlight` I've removed but forgot to push